common/djangoapps/student/models.py

```diff
@@ -939,6 +939,11 @@ def __repr__(self):
         if self.lockout_until is not None:
             date_str = self.lockout_until.isoformat()
 
+        try:
```

Comment:
There's a logic duplication that is not really necessary between the `__str__` and `__repr__` methods:

Please use this variant instead:

This compatible with both Python 2 which expects `str` (that is `bytes` in Python 3) and Python 3 expects its new `str` (that is `unicode` in Python 2). See this [Stackoverflow post for more details](https://consideratecode.com/2017/08/25/what-does-python_2_unicode_compatible-do/). Also the [article that explains the `@python_2_unicode_compatible` functionality](https://consideratecode.com/2017/08/25/what-does-python_2_unicode_compatible-do/).

```
def __repr__(self):
    """
    Repr -> LoginFailures(Username: count - date)
    """
    return 'LoginFailures({self_string})'.format(
        self_string=str(self),
    )
```

Reply:
Good point, I removed this method.